handler/core/src/main/java/io/knotx/fragments/handler/FragmentsHandler.java

```diff
@@ -125,10 +126,18 @@ private RequestEvent copyRequestEvent(RequestEvent requestEvent) {
   }
 
   private List<Fragment> retrieveFragments(List<FragmentEvent> events,
-      Predicate<FragmentEvent> predicate) {
+                                           Predicate<FragmentEvent> predicate) {
     return events.stream()
         .filter(predicate)
         .map(FragmentEvent::getFragment)
         .collect(Collectors.toList());
   }
+
+  private boolean shouldPassInvalidFragments(RequestContext requestContext) {
+    ClientRequest request = requestContext.getRequestEvent().getClientRequest();
```

Comment:
Please provide unit tests to validate:
- no `allowInvalidFragmentsParam` is configured but passed in request
- `allowInvalidFragmentsParam` is configured and passed in request
- no `allowInvalidFragmentsHeader` is configured but passed in request
- `allowInvalidFragmentsHeader` is configured and passed in request